lib/loadMtl.js

```diff
@@ -217,11 +217,19 @@ function loadMaterialTexture(material, name, texturePath, textureOptions, mtlDir
 
     var texturePromise = texturePromiseMap[texturePath];
     if (!defined(texturePromise)) {
+        var shallowPath = path.resolve(path.join(mtlDirectory, path.basename(texturePath)));
         if (options.secure && outsideDirectory(texturePath, mtlDirectory)) {
-            options.logger('Could not read texture file at ' + texturePath + ' because it is outside of the mtl directory and the secure flag is true. This texture will be ignored.');
-            texturePromise = Promise.resolve();
+            // Try looking for the texture in the same directory as the obj
+            texturePromise = loadTexture(shallowPath, textureOptions)
+                .catch(function() {
+                    options.logger('Could not read texture file at ' + texturePath + ' because it is outside of the mtl directory and the secure flag is true. This texture will be ignored.');
```

Comment:
> > Also not sure this should happen as a generic catch, since loadTexture can also reject due to a pngjs or jpegjs error. So technically this could cause the wrong texture to be loaded if the intended texture fails due to decode-related problems, although that'd probably be a pretty contrived case.

> I also considered this but didn't think it would ever really happen.

Agreed that the case described above is unlikely, but, using generic `catch`es like this right now can also make the logged output message inaccurate, which makes the input harder for the user to fix.

For example, this message could log either because the texture couldn't be found at all (described), because it was corrupted, or because it was found in the root but was corrupted.

Reply:
Maybe sufficient to log the error in addition to a more general warning for `basename(texturePath)`?